components/renku_data_services/k8s_watcher/__init__.py

```diff
@@ -1 +1,100 @@
 """K8s watcher and k8s clients."""
+
+from __future__ import annotations
```

Comment:
since this is business logic, I'd but it in core.py to be consistent. __init__.py is a good place for package level initialization code (as it's processed on every import) and for exposing easier imports via __all__, but things like business logic should not be in __init__.py.

Reply:
@Panaetius we discussed this but I did not want to make even larger refactoring changes during the build. We will do what you propose for sure during cooldown.

Reply:
@sambuc when we discussed this what I had in mind is to just import `K8sWatcher` in `__init__.py`. Then it can be imported from the module. See for example this: https://github.com/SwissDataScienceCenter/renku-data-services/blob/main/components/renku_data_services/errors/__init__.py#L3

Sor you can import errors just like `from renku_data_services.errors import BaseError` rather than `from renku_data_services.errors.errors import BaseError`

Reply:
I have moved the content of `__init__.py` `core.py` and added import + re-export so that public objects appear at the main module level.

I also fixed a couple of thing to be more explicit about testing `None` values and small refactors.

I will submit later in the day the refactor to use an enum instead of lower case strings for the `Kinds` being cached.